vault/resource_plugin.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+const (
+	fieldSHA256   = "sha256"
+	fieldCommand  = "command"
+	fieldArgs     = "args"
+	fieldEnv      = "env"
+	fieldOCIImage = "oci_image"
+	fieldRuntime  = "runtime"
+)
+
+func pluginResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: pluginWrite,
```

Comment:
Should there be any version checks here? My guess is no, since registering plugins has been in Vault for a while, but just wanted to confirm

Reply:
Good call out. Registering plugins in general has been available even pre 1.0, so I don't think it's worth it. However, there are some parameters (versions and containerised plugins) that need more recent version support. I added some checks around those parameters.